### ISSUE TITLE
Read-only opening of devices

### ISSUE THREAD

**Issue by sim642** [gvalkov/python-evdev#35]
Currently InputDevice wants to open the device `os.RDWR` ([code](https://github.com/gvalkov/python-evdev/blob/master/evdev/device.py#L93)) but there's good reason to be able to open a device `os.RDONLY`, either if there's no intention to modify anything or there's not sufficient permissions.

Making a custom derived class with changed `__init__` seems to work but I think it could very well be included in the library:
```py
from evdev import InputDevice, _input, DeviceInfo
import os

class ReadInputDevice(InputDevice):
	def __init__(self, dev):
		'''
		:param dev: path to input device
		'''

		#: Path to input device.
		self.fn = dev

		#: A non-blocking file descriptor to the device file.
		self.fd = os.open(dev, os.O_RDONLY | os.O_NONBLOCK)

		# Returns (bustype, vendor, product, version, name, phys, capabilities).
		info_res = _input.ioctl_devinfo(self.fd)

		#: A :class:`DeviceInfo <evdev.device.DeviceInfo>` instance.
		self.info = DeviceInfo(*info_res[:4])

		#: The name of the event device.
		self.name = info_res[4]

		#: The physical topology of the device.
		self.phys = info_res[5]

		#: The evdev protocol version.
		self.version = _input.ioctl_EVIOCGVERSION(self.fd)

		#: The raw dictionary of device capabilities - see `:func:capabilities()`.
		self._rawcapabilities = _input.ioctl_capabilities(self.fd)

		#: The number of force feedback effects the device can keep in its memory.
		self.ff_effects_count = _input.ioctl_EVIOCGEFFECTS(self.fd)

```

**Comment by sim642** [comment]
Same would apply to `list_devices` which only lists readable **and** writable devices while I might not actually require writability.

**Comment by gvalkov** [comment]
That is quite reasonable. I'll definitely look into adding it. Thanks.

**Comment by gvalkov** [comment]
Appologies for the delay. Fixed in 4051ba9c7c731566b33251aed149f103aa8f7256 and available in `evdev>=0.5.0`.
